src/java.base/unix/classes/sun/nio/ch/PollSelectorImpl.java

```diff
@@ -120,7 +120,7 @@ protected int doSelect(Consumer<SelectionKey> action, long timeout)
                 if (numPolled == IOStatus.INTERRUPTED && timedPoll) {
                     // timed poll interrupted so need to adjust timeout
                     long adjust = System.nanoTime() - startTime;
-                    to -= TimeUnit.MILLISECONDS.convert(adjust, TimeUnit.NANOSECONDS);
+                    to -= (int)TimeUnit.MILLISECONDS.convert(adjust, TimeUnit.NANOSECONDS);
```

Comment:
This one can also be changed to:

`to =- (int) TimeUnit.NANOSECONDS.toMillis(adjust);`